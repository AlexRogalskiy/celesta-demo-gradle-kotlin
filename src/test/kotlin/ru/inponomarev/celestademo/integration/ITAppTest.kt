package ru.inponomarev.celestademo.integration

import org.approvaltests.Approvals
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Tag
import org.junit.jupiter.api.Test
import retrofit2.Retrofit
import retrofit2.converter.scalars.ScalarsConverterFactory

@Tag("integration")
class ITAppTest {
    private val retrofit = Retrofit.Builder()
        .baseUrl("http://localhost:8088/api/v1/")
        .addConverterFactory(ScalarsConverterFactory.create())
        .build()
    private val api = retrofit.create(AppApi::class.java)

    @Test
    fun getItems() {
        val response = api.items.execute()
        assertThat(response.isSuccessful).isTrue
        Approvals.verifyJson(response.body())
    }
}