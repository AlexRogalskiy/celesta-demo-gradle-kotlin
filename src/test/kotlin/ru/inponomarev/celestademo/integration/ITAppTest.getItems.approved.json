[
  {
    "id": "123",
    "name": "cheese",
    "orderedQuantity": 10
  }
]